.github/codeql/codeql-config.yml

```diff
@@ -0,0 +1,12 @@
+name: "CodeQL Security Analysis"
+
+paths-ignore:
+  - core/tests/data
+  # This file causes CodeQL to throw the error "Use of a broken or weak
+  # cryptographic hashing algorithm on sensitive data," (line 568), but sha1 is
+  # not used for enciphering so preimage attacks are not a risk.
+  - core/utils.py
```

Comment:
I think we should also add a TODO for this.

Reply:
Done